File: Vireo3/test/org/tdl/vireo/model/MockCustomActionDefinition.java
package org.tdl.vireo.model;

/**
 * This is a simple mock custom action definition class that may be useful for testing. Feel free to extend this to add in extra parameters that you feel appropriate.
 * 
 * The basic concept is all properties are public so you can create the mock object and then set whatever relevant properties are needed for your particular test.
 * 
 * @author <a href="http://www.scottphillips.com">Scott Phillips</a>
 */
public class MockCustomActionDefinition extends AbstractMock implements CustomActionDefinition {

	/* Custom Action Definition Properties */
	public int displayOrder;
	public String label;
	private Boolean isStudentVisible;

	@Override
	public MockCustomActionDefinition save() {
		return this;
	}

	@Override
	public MockCustomActionDefinition delete() {
		return this;
	}

	@Override
	public MockCustomActionDefinition refresh() {
		return this;
	}

	@Override
	public MockCustomActionDefinition merge() {
		return this;
	}

	@Override
	public MockCustomActionDefinition detach() {
		return this;
	}

	@Override
	public int getDisplayOrder() {
		return displayOrder;
	}

	@Override
	public void setDisplayOrder(int displayOrder) {
		this.displayOrder = displayOrder;
	}

	@Override
	public String getLabel() {
		return label;
	}

	@Override
	public void setLabel(String label) {
		this.label = label;
	}

	@Override
	public Boolean isStudentVisible() {
		return isStudentVisible;
	}

	@Override
	public void setIsStudentVisible(Boolean isStudentVisible) {
		this.isStudentVisible = isStudentVisible;
	}

}

File: src/main/webapp/tests/mocks/models/mockCustomActionDefinition.js
var dataCustomActionDefinition1 = {
    id: 1,
    isStudentVisible: false,
    label: "custom action 1"
};

var dataCustomActionDefinition2 = {
    id: 2,
    isStudentVisible: false,
    label: "custom action 2"
};

var dataCustomActionDefinition3 = {
    id: 3,
    isStudentVisible: true,
    label: "custom action 3"
};

var dataCustomActionDefinition4 = {
    id: 4,
    isStudentVisible: true,
    label: "custom action 4"
};

var dataCustomActionDefinition5 = {
    id: 5,
    isStudentVisible: true,
    label: "custom action 5"
};

var dataCustomActionDefinition6 = {
    id: 6,
    isStudentVisible: false,
    label: "custom action 6"
};

var mockCustomActionDefinition = function($q) {
    var model = mockModel($q, dataCustomActionDefinition1);

    return model;
};

angular.module('mock.customActionDefinition', []).service('CustomActionDefinition', mockCustomActionDefinition);

